dev: add flag for extra docker args

herkolategan opened the pull request:
In order to support git alternates an additional volume must be supplied to the
bazel support docker. The TeamCity script already supplies a utility method
called `run_bazel` which supplies the volume to support git alternates.

This change allows for extra arguments to be passed via a command line argument
to the docker invocation that is utilised for cross builds by the `dev` utility.

Resolves: #101829

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/101931)
<!-- Reviewable:end -->

rickystewart commented:
For `dev` we typically just use command-line arguments instead of environment variables. Any particular reason why we have not done that here?

herkolategan reviewed (commented):
My reasoning was that it seems the docker invocation could form part of multiple commands now and in the future, thus an environment variable would ensure being able to always supply additional arguments to the invocation. But that said, I'm not opposed to sticking to command-line arguments to keep things uniform.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/101931#-:-NTYIh5Y505yYPu83zo_:b-ey3vql)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @smg260 and @srosenberg)



<!-- Sent from Reviewable.io -->

herkolategan committed:
dev: add flag for extra docker args

In order to support git alternates an additional volume must be supplied to the
bazel support docker. The TeamCity script already supplies a utility method
called `run_bazel` which supplies the volume to support git alternates.

This change allows for extra arguments to be passed via a command line argument
to the docker invocation that is utilised for cross builds by the `dev` utility.

herkolategan reviewed (commented):
@rickystewart Updated my PR to take command-line arguments instead.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/101931#-:-NTre4B-1LAwVeKVaNa7:b-4z1igi)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @smg260 and @srosenberg)



<!-- Sent from Reviewable.io -->

srosenberg reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/101931#-:-NTutRuE3CsxY460iObe:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @rickystewart and @smg260)



<!-- Sent from Reviewable.io -->

rickystewart reviewed (approved):
LGTM. Consider bumping the version number in the top-level `./dev` script so people pick up the change.

herkolategan commented:
TFTR!
bors r=srosenberg,rickystewart
